### ISSUE TITLE
why the dice is 0 or nan

### ISSUE THREAD

**Issue by 2345678y3** [MIC-DKFZ/nnUNet#2034]
![屏幕截图 2024-03-24 090414](https://github.com/MIC-DKFZ/nnUNet/assets/100781353/bad03582-e749-4a53-aece-357b66e6b514)
thank you

**Comment by MD7sjh** [comment]
I have encountered your problem before, can you describe your situation in detail.

**Comment by 2345678y3** [comment]
My dataset is segmentation of brain metastases, and my labels may have 1 2 3 ... The maximum number of labels is 42, the size of the label voxels is different, the dataset size is 344 cases, the loss cannot converge to -0.1 after training with nnunetv2, and there are many 0s and nans in each enpoch and in dice, and the final training of the model，the dice
![progress](https://github.com/MIC-DKFZ/nnUNet/assets/100781353/e1169f8b-75c8-4c53-8189-5b72ccd3eef3)
 is only 0.03

**Comment by 2345678y3** [comment]
> I have encountered your problem before, can you describe your situation in detail.↳

What is the reason for your situation?thank you

**Comment by 2345678y3** [comment]
I have tried the nnUNetTrainerDiceCELoss_noSmooth  but  the problem hasn't been solved yet.

**Comment by Karol-G** [comment]
Dear @2345678y3,

Thank you for using nnU-Net.

To assist you better, we need a clearer understanding of the issue you're encountering. It appears there might be some confusion regarding your problem or the specific outcomes you aim to achieve with nnU-Net.

Could you please provide a more detailed description of your task and the problems you have encountered? This information will greatly aid us in identifying potential solutions.

In addition, we require the following specific information to proceed with a thorough investigation:

- The dataset.json file and the exact preprocessing, train, and inference commands you used.
- Training images and their respective ground truth. For confidentiality and review purposes, you can post screenshots or send full examples to karol.gotkowski@dkfz.de via cloud storage. Rest assured, any data received will be treated with the utmost confidentiality and will be deleted after the review.
- The training log and the progress.png for every fold, compiled into a zip archive.

These details are crucial for us to understand the context and specifics of your setup, which in turn will help us provide you with targeted support.

We appreciate your cooperation and look forward to your response.

Best regards,

Karol Gotkowski

**Comment by 2345678y3** [comment]
杨钦
***@***.***



&nbsp;this is my training code and the result,thank you&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "MIC-DKFZ/nnUNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年4月2日(星期二) 下午4:24
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [MIC-DKFZ/nnUNet] why the dice is 0 or nan (Issue #2034)





 
Dear @2345678y3,
 
Thank you for using nnU-Net.
 
To assist you better, we need a clearer understanding of the issue you're encountering. It appears there might be some confusion regarding your problem or the specific outcomes you aim to achieve with nnU-Net.
 
Could you please provide a more detailed description of your task and the problems you have encountered? This information will greatly aid us in identifying potential solutions.
 
In addition, we require the following specific information to proceed with a thorough investigation:
  
The dataset.json file and the exact preprocessing, train, and inference commands you used.
 
Training images and their respective ground truth. For confidentiality and review purposes, you can post screenshots or send full examples to ***@***.*** via cloud storage. Rest assured, any data received will be treated with the utmost confidentiality and will be deleted after the review.
 
The training log and the progress.png for every fold, compiled into a zip archive.
  
These details are crucial for us to understand the context and specifics of your setup, which in turn will help us provide you with targeted support.
 
We appreciate your cooperation and look forward to your response.
 
Best regards,
 
Karol Gotkowski

**Comment by Karol-G** [comment]
Hey,

You did not send any additional information. Please review your reply again. Thank you.

Best regards,
Karol

**Comment by MD7sjh** [comment]
Have you ever changed your network architecture？My situation is I change the network, it can not read my dataset properly in my inference stage.

**Comment by 2345678y3** [comment]
> Hey,
> 
> You did not send any additional information. Please review your reply again. Thank you.
> 
> Best regards, Karol

i have sent it to you again，could  you take a look again? thank you !

**Comment by 2345678y3** [comment]
> Hey,
> 
> You did not send any additional information. Please review your reply again. Thank you.
> 
> Best regards, Karol

[mynnuet.zip](https://github.com/MIC-DKFZ/nnUNet/files/14914347/mynnuet.zip)

**Comment by deepsea920415** [comment]
I also saw same result.
Dice was 0.01 and and see too man "nan" at the result.

**Comment by Karol-G** [comment]
Hey @2345678y3,

According to your dataset.json your dataset has 42 different tumor classes. Are you sure this is correct? A class label separates usually different concepts such as different organs or different regions of the same tumor. It seems to me that your dataset has in total 42 tumor instances and that you assigned a class to each instance. Can you explain your task a bit more and comment on if my assumption is right?

Best,
Karol

**Comment by 2345678y3** [comment]
Thank you for your answer，sir .My dataset is brain metastasis ,so,one patient &nbsp;may have more than one metastasises that from the same tumor in his brain,thus,if the person has 42 metastasises in his brain,i will label every metastasis &nbsp;with different labelname。so i should label them with the same label right?



发自我的iPhone


------------------ Original ------------------
From: Karol Gotkowski ***@***.***&gt;
Date: Mon,May 6,2024 8:01 PM
To: MIC-DKFZ/nnUNet ***@***.***&gt;
Cc: 2345678y3 ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [MIC-DKFZ/nnUNet] why the dice is 0 or nan (Issue #2034)

**Comment by Karol-G** [comment]
@deepsea920415 Please open a separate issue for your problem. It is different to this issue opened by @2345678y3 . Thank you.

**Comment by deepsea920415** [comment]
Already done.(CBCT teeth segmentation)
But couldn't get any good idea.

**Comment by Karol-G** [comment]
Hey @2345678y3,

Yes, exactly nnunet is not capable of doing instance segmentation. It can only do semantic segmentation. So in your case, it seems there is only one class which would be the class "Metastasis".

Best,
Karol

**Comment by 2345678y3** [comment]
thank you for your help &nbsp;sir



发自我的iPhone


------------------ Original ------------------
From: Karol Gotkowski ***@***.***&gt;
Date: Mon,May 6,2024 8:47 PM
To: MIC-DKFZ/nnUNet ***@***.***&gt;
Cc: 2345678y3 ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [MIC-DKFZ/nnUNet] why the dice is 0 or nan (Issue #2034)





 
Hey @2345678y3,
 
Yes, exactly nnunet is not capable of doing instance segmentation. It can only do semantic segmentation. So in your case, it seems there is only one class which would be the class "Metastasis".
 
Best,
 Karol

**Comment by yibeizixia** [comment]
Have you solved the issue yet? I'm facing the same problem as well.

**Comment by Karol-G** [comment]
@yibeizixia Please open a separate issue for your problem as it is most likely not related to @2345678y3
